device.yml

```
@@ -0,0 +1,273 @@
+    maskType: TriggerInput
+    description: Emits a start software-trigger on the channels specified in the mask.
+  StopSoftwareTrigger:
+    <<: *softtrigger
+    address: 61
+    maskType: TriggerInput
+    description: Emits a stop software-trigger on the channels specified in the mask.
+  ArmPwmChannels:
+    address: 62
+    type: U8
+    access: Write
+    maskType: PwmChannels
+    description: Arms the PWM channels specified in the mask. Once a PWM is triggered, it must be rearmed.
+  Reserved0: &reserved
+    address: 63
+    type: U8
+    access: Read
+    visibility: private
+    description: Reserved for future use.
+  Trigger0Mode: &triggermode
+    address: 64
+    type: U8
+    access: Write
+    payloadSpec:
+      TriggerMode:
+        maskType: TriggerModeConfig
+        description: Specifies the trigger mode.
+        mask: 0x03
+      Polarity:
+        maskType: TriggerPolarity
+        description: Specifies the polarity of the trigger signal.
+        mask: 0x10
+    description: Trigger mode of input channel 0.
+  Trigger1Mode:
+    <<: *triggermode
+    address: 65
+    description: Trigger mode of input channel 1.
+  Trigger2Mode:
+    <<: *triggermode
+    address: 66
+    description: Trigger mode of input channel 2.
+  Trigger3Mode:
+    <<: *triggermode
+    address: 67
+    description: Trigger mode of input channel 3.
+  RequestEnable:
+    address: 68
+    type: U8
+    access: Write
+    maskType: PwmChannels
+    description: If set to Low, it will bypass the EnablePwmChannels and ArmPwmChannels register function.
+  EnablePwmChannels:
+    address: 69
+    type: U8
+    access: Write
+    maskType: PwmChannels
+    description: Enable enable
+  TriggerAllMode:
+    address: 70
+    type: U8
+    access: Write
+    payloadSpec:
+      TriggerMode:
+        maskType: TriggerAllModeConfig
+        description: Specifies the trigger mode.
+        mask: 0x03
+      Polarity:
+        maskType: TriggerPolarity
+        description: Specifies the polarity of the trigger signal.
+        mask: 0x10
+    description: When triggered, all channels will be affected by the event.
+  TriggerChannelState:
+    address: 71
+    type: U8
+    access: Read
+    maskType: TriggerInput
+    description: Current state of all trigger channel inputs.
+  PwmChannelState:
+    address: 72
+    type: U8
+    access: Read
+    maskType: PwmChannels
+    description: Current state of all PWM channel outputs.
+  PwmState:
+    address: 73
+    type: U8
+    access: Event
+    maskType: PwmChannels
+    description: The state of the PWM for each channel. Emits an event each time a PMW starts or stops.
+  EnableEvents:
+    address: 74
+    type: U8
+    access: Write
+    maskType: MultiPwmEvents
+    description: Enables the generation of events.
+bitMasks:
+  PwmChannels:
+    description: Available PWM output channels.
+    bits:
+      Channel0: 0x01
+      Channel1: 0x02
+      Channel2: 0x04
+      Channel3: 0x08
+  TriggerInput:
```

```suggestion
  TriggerInputs:
```

In line with [naming conventions for bitmask flags](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1714), this should be pluralized.